osf/models/user.py

```diff
@@ -972,14 +972,7 @@ def deactivate_account(self):
                 user_id=self._id,
                 username=self.username
             )
-        except mailchimp_utils.mailchimp.ListNotSubscribedError:
-            pass
-        except mailchimp_utils.mailchimp.InvalidApiKeyError:
-            if not website_settings.ENABLE_EMAIL_SUBSCRIPTIONS:
-                pass
-            else:
-                raise
```

Comment:
One side effect: this no longer raises an error. Realistically, `ENABLE_EMAIL_SUBSCRIPTIONS` should likely be equivalent to `bool(MAILCHIMP_API_KEY)`